Mathlib/Topology/Algebra/Module/Alternating.lean

```diff
@@ -0,0 +1,616 @@
+@[simps]
+def toMultilinearAddHom : Λ^ι⟮R; M; N⟯ →+ ContinuousMultilinearMap R (fun _ : ι => M) N :=
+  ⟨⟨fun f => f.1, rfl⟩, fun _ _ => rfl⟩
+
+end ContinuousAdd
+
+/-- If `f` is a continuous alternating map, then `f.to_continuous_linear_map m i` is the continuous
+linear map obtained by fixing all coordinates but `i` equal to those of `m`, and varying the
+`i`-th coordinate. -/
+@[simps! apply]
+def toContinuousLinearMap [DecidableEq ι] (m : ι → M) (i : ι) : M →L[R] N :=
+  f.1.toContinuousLinearMap m i
+
+/-- The cartesian product of two continuous alternating maps, as a continuous alternating map. -/
+@[simps!]
+def prod (f : Λ^ι⟮R; M; N⟯) (g : Λ^ι⟮R; M; N'⟯) : Λ^ι⟮R; M; N × N'⟯ :=
+  ⟨f.1.prod g.1, (f.toAlternatingMap.prod g.toAlternatingMap).map_eq_zero_of_eq⟩
+
+/-- Combine a family of continuous alternating maps with the same domain and codomains `M' i` into a
+continuous alternating map taking values in the space of functions `Π i, M' i`. -/
+def pi {ι' : Type _} {M' : ι' → Type _} [∀ i, AddCommMonoid (M' i)] [∀ i, TopologicalSpace (M' i)]
+    [∀ i, Module R (M' i)] (f : ∀ i, Λ^ι⟮R; M; M' i⟯) : Λ^ι⟮R; M; ∀ i, M' i⟯ :=
+  ⟨ContinuousMultilinearMap.pi fun i => (f i).1,
+    (AlternatingMap.pi fun i => (f i).toAlternatingMap).map_eq_zero_of_eq⟩
+
+@[simp]
+theorem coe_pi {ι' : Type _} {M' : ι' → Type _} [∀ i, AddCommMonoid (M' i)]
+    [∀ i, TopologicalSpace (M' i)] [∀ i, Module R (M' i)] (f : ∀ i, Λ^ι⟮R; M; M' i⟯) :
+    ⇑(pi f) = fun m j => f j m :=
+  rfl
+
+theorem pi_apply {ι' : Type _} {M' : ι' → Type _} [∀ i, AddCommMonoid (M' i)]
+    [∀ i, TopologicalSpace (M' i)] [∀ i, Module R (M' i)] (f : ∀ i, Λ^ι⟮R; M; M' i⟯) (m : ι → M)
+    (j : ι') : pi f m j = f j m :=
+  rfl
+
+section
+
+variable (R M)
+
+/-- The evaluation map from `ι → M₂` to `M₂` is alternating at a given `i` when `ι` is subsingleton.
+-/
+@[simps! toContinuousMultilinearMap apply]
+def ofSubsingleton [Subsingleton ι] (i' : ι) : Λ^ι⟮R; M; M⟯ :=
+  { AlternatingMap.ofSubsingleton R _ i' with
+    toContinuousMultilinearMap := ContinuousMultilinearMap.ofSubsingleton R _ i' }
+
+@[simp]
+theorem ofSubsingleton_toAlternatingMap [Subsingleton ι] (i' : ι) :
+    (ofSubsingleton R M i').toAlternatingMap = AlternatingMap.ofSubsingleton R M i' :=
+  rfl
+
+variable (ι)
+
+/-- The constant map is alternating when `ι` is empty. -/
+@[simps! toContinuousMultilinearMap apply]
+def constOfIsEmpty [IsEmpty ι] (m : N) : Λ^ι⟮R; M; N⟯ :=
+  { AlternatingMap.constOfIsEmpty R M ι m with
+    toContinuousMultilinearMap := ContinuousMultilinearMap.constOfIsEmpty R (fun _ => M) m }
+
+@[simp]
+theorem constOfIsEmpty_toAlternatingMap [IsEmpty ι] (m : N) :
+    (constOfIsEmpty R M ι m).toAlternatingMap = AlternatingMap.constOfIsEmpty R M ι m :=
+  rfl
+
+end
+
+/-- If `g` is continuous alternating and `f` is a collection of continuous linear maps,
+then `g (f₁ m₁, ..., fₙ mₙ)` is again a continuous alternating map, that we call
```

Comment:
The docstring is not correct, you need to compose with the same map at all indices.